clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp

```diff
@@ -29,6 +29,12 @@ void tooling::dependencies::configureInvocationForCaching(
   // Clear this otherwise it defeats the purpose of making the compilation key
   // independent of certain arguments.
   CI.getCodeGenOpts().DwarfDebugFlags.clear();
+  if (FrontendOpts.ProgramAction == frontend::GeneratePCH) {
+    // Clear paths that are emitted into binaries; they do not affect PCH.
+    // For modules this is handled in ModuleDepCollector.
```

Comment:
Would it make sense to strip these for PCHs in clang-scan-deps too?

Reply:
I think we could, but I can't think of a reason to make such a change without caching.  If you can think of a reason to do it then I'm happy to move this to DependencyScanningWorker and do it upstream.

Reply:
I guess one tiny advantage is shorter command lines and having the behavior unified. But I don't have strong preferences, up to you.

Reply:
We talked about this offline; it might be interesting in the future to do a more holistic audit of options we could strip from PCH (or even TU) command-lines, but not worthwhile for just these two options.